src/main/groovy/net/neoforged/gradleutils/spotless/SpotlessUtilsExtension.groovy

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.gradleutils.spotless
+
+import com.diffplug.gradle.spotless.JavaExtension
+import com.diffplug.gradle.spotless.SpotlessExtension
+import com.diffplug.gradle.spotless.SpotlessTask
+import groovy.transform.CompileStatic
+import org.gradle.api.Project
+
+import javax.inject.Inject
+import java.nio.file.Files
+
+@CompileStatic
+abstract class SpotlessUtilsExtension {
+    final File configPath
+
+    @Inject
+    SpotlessUtilsExtension(Project project) {
+        configPath = new File(project.rootDir, '.gradle/formatter-config.xml')
+
+        if (!configPath.exists()) {
+            Files.createDirectories(configPath.toPath().parent)
+
+            try (
+                final input = SpotlessUtilsExtension.getResourceAsStream('/formatter-config.xml');
+                final output = Files.newOutputStream(configPath.toPath())
+            ) {
+                byte[] buffer = new byte[1024];
+                int len
+                while ((len = input.read(buffer)) > 0) {
+                    output.write(buffer, 0, len)
+                }
+            }
+        }
+
+        final extract = project.tasks.register('extraGUSpotlessConfiguration', ExtractSpotlessConfiguration) {
+            it.output.set(configPath)
+        }
+
+        project.tasks.withType(SpotlessTask).configureEach {
+            it.dependsOn(extract)
+        }
```

Comment:
Is it possible in some fashion to create an extension property which controls the formatter config extraction?

Reply:
Control what part of that?

Reply:
Extraction is what this entire plugin does. If you don't want to extract the configuration and use the GU-provided config, then the plugin is useless.